src/IMangrove.sol

```diff
@@ -62,118 +101,334 @@ interface IMangrove is HasMgvEvents {
+  ///@dev See also `marketOrderByVolume`.
+  function marketOrderForByVolume(OLKey memory olKey, uint takerWants, uint takerGives, bool fillWants, address taker)
+    external
+    returns (uint takerGot, uint takerGave, uint bounty, uint feePaid);
+
   // # Cleaning functions
 
+  /* # Cleaning */
+  ///@notice Cleans multiple offers, i.e. executes them and remove them from the book if they fail, transferring the failure penalty as bounty to the caller.
+  ///@param olKey The offer list key given by (maker) `outbound`, (maker) `inbound`, and `tickSpacing`.
+  ///@param targets The offers to clean, identified by their (`offerId, tick, gasreq, takerWants`) that will make them fail.
+  ///@param taker The taker used for transfers (should be able to deliver token amounts).
+  ///@return successes The number of successfully cleaned offers.
+  ///@return bounty The total bounty received by the caller.
+  ///@dev If an offer succeeds, the execution of that offer is reverted, it stays in the book, and no bounty is paid; The `cleanByImpersonation` function itself will not revert.
+  ///@dev Note that Mangrove won't attempt to execute an offer if the values in a target don't match its offer. To distinguish between a non-executed clean and a fail clean (due to the offer itself not failing), you must inspect the log (see `MgvLib.sol`) or check the received bounty.
+  ///@dev Any `taker` can be impersonated when cleaning because:
+  ///@dev - The function reverts if the offer succeeds, reverting any token transfers.
+  ///@dev - After a `clean` where the offer has failed, all ERC20 token transfers have also been reverted -- but the sender will still have received the bounty of the failing offers. */
   function cleanByImpersonation(OLKey memory olKey, MgvLib.CleanTarget[] calldata targets, address taker)
     external
     returns (uint successes, uint bounty);
 
   // # Maker functions
 
-  receive() external payable;
-
+  ///@notice Adds funds to Mangrove for the caller (the maker) to use for provisioning offers.
   function fund() external payable;
 
+  ///@notice Adds funds to Mangrove for the caller (the maker) to use for provisioning offers.
+  receive() external payable;
+
+  ///@notice Adds funds to Mangrove for the maker to use for provisioning offers.
+  ///@param maker The maker to add funds for.
   function fund(address maker) external payable;
 
+  ///@notice Withdraws the caller's (the maker's) native token (funds for provisioning offers) by transferring them to the caller.
+  ///@param amount the amount to withdraw.
+  ///@return noRevert whether the transfer succeeded.
   function withdraw(uint amount) external returns (bool noRevert);
 
+  ///@notice Gets the maker's balance of native tokens (funds for provisioning offers).
+  ///@param maker The maker to get the balance for.
+  ///@return balance The maker's balance of native tokens (funds for provisioning offers).
```

Comment:
Maybe we should clarify that the returned value is the "free" balance, i.e., not including any provision tied up by live offers.